### ISSUE TITLE
[HOLD for payment 2024-07-24] [#Wave-Control - Add Sage Intacct] Settings Configuration in NewDot: Export

### ISSUE THREAD

**Issue by yuwenmemon** [Expensify/App#43534]
Tracking GH: https://github.com/Expensify/Expensify/issues/388780

Design doc section: https://docs.google.com/document/d/1k3ZFw8KB55yPUSCG6KYZlwpwEtmRt3eUshwxs7bZq5I/edit#heading=h.vn04szhrriv7

---

You will find in the following issue the pseudocode for the different flows. Use them as a guide in your implementation.

### Main page

```JS
function SageIntacctExportPage({policy}: WithPolicyProps) {
    const {export: exportConfig, pendingFields, errorFields} = policy?.connections?.intacct?.config ?? {};

    const currentSageIntacctOrganizationName = useMemo(() => getCurrentSageIntacctOrganizationName(policy ?? undefined), [policy]);

    const sections = useMemo(
        () => [
            {
                description: translate('workspace.sageIntacct.preferredExporter'),
                action: () => Navigation.navigate(ROUTES.POLICY_ACCOUNTING_SAGE_INTACCT_PREFERRED_EXPORTER.getRoute(policyID)),
                title: exportConfig.exporter || translate('workspace.sageIntacct.notConfigured'),
                hasError: !!errorFields?.export.exporter,
                pendingAction: pendingFields?.export.exporter,
            },
            {
                description: translate('workspace.sageIntacct.exportDate'),
                action: () => Navigation.navigate(ROUTES.POLICY_ACCOUNTING_SAGE_INTACCT_EXPORT_DATE.getRoute(policyID)),
                title: translate('workspace.sageIntacct.exportDate'),
                hasError: !!errorFields?.export.exportDate,
                pendingAction: pendingFields?.export.exportDate,
            },
            {
                description: translate('workspace.sageIntacct.exportReimbursableExpensesAs'),
                action: () => Navigation.navigate(ROUTES.POLICY_ACCOUNTING_SAGE_INTACCT_REIMBURSABLE_EXPENSES.getRoute(policyID)),
                title: exportConfig.reimbursable ? translate(`workspace.accounting.reimbursable.${exportConfig.reimbursable}`) : translate('workspace.sageIntacct.notConfigured'),
                hasError: !!errorFields?.export.reimbursable,
                pendingAction: pendingFields?.export.reimbursable,
            },
            {
                description: translate('workspace.sageIntacct.exportNonReimbursableExpensesAs'),
                action: () => Navigation.navigate(ROUTES.POLICY_ACCOUNTING_SAGE_INTACCT_NON_REIMBURSABLE_EXPENSES.getRoute(policyID)),
                title: exportConfig.nonReimbursable ? translate(`workspace.accounting.nonReimbursable.${exportConfig.nonReimbursable}`) : translate('workspace.sageIntacct.notConfigured'),
                hasError: !!errorFields?.export.nonReimbursable,
                pendingAction: pendingFields?.export.nonReimbursable,
            },
        ],
        [
		//... useMemo() dependencies
        ],
    );

    return (
        <ConnectionLayout
            displayName={SageIntacctExportPage.displayName}
            headerTitle="workspace.accounting.export"
            headerSubtitle={currentSageIntacctOrganizationName}
            accessVariants={[CONST.POLICY.ACCESS_VARIANTS.ADMIN, CONST.POLICY.ACCESS_VARIANTS.PAID]}
            policyID={policyID}
            featureName={CONST.POLICY.MORE_FEATURES.ARE_CONNECTIONS_ENABLED}
            contentContainerStyle={styles.pb2}
            titleStyle={styles.ph5}
            connectionName={CONST.POLICY.CONNECTIONS.NAME.SAGE_INTACCT}
        >
	                <Text style={[styles.ph5, styles.pb5]}>{translate('workspace.sageIntacct.exportDescription')}</Text>
            {sections.map((section) => (
                <OfflineWithFeedback key={section.description} pendingAction={section.pendingAction}>
                    <MenuItemWithTopDescription
                        title={section.title}
                        description={section.description}
                        shouldShowRightIcon
                        onPress={section.action}
                        brickRoadIndicator={section.hasError ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : undefined}
                    />
                </OfflineWithFeedback>
            ))}
        </ConnectionLayout>
    );
}

SageIntacctExportPage.displayName = 'PolicySageIntacctExportPage';

export default withPolicy(SageIntacctExportPage);
```
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017260ab59e4fe16a6</li>
        <li>Upwork Job ID: 1800760782833912827</li>
        <li>Last Price Increase: 2024-06-12</li>
<li>Automatic offers: </li>
<ul>
<li>hungvu193 | Contributor | 102699116</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @puneetlath</details>

**Comment by SzymczakJ** [comment]
Hey! I’m Jakub Szymczak from Software Mansion, an expert agency, and I’d like to work on this issue! I might find somebody from SWM to help me with that, so in that case we can reassign. But in the meantime feel free to assign me 😄

**Comment by SzymczakJ** [comment]
Update:
Haven't started working on it. I'll probably find somebody from Software Mansion to work on this.

**Comment by hungvu193** [comment]
Cool. Still waiting for the PR...

**Comment by war-in** [comment]
Hi, I'm Marcin from Software Mansion, an expert agency, and I'd like to work on this issue :)

**Comment by war-in** [comment]
Small update here :) [This is the draft PR](https://github.com/Expensify/App/pull/43776) I'll continue working on Monday. 
Currently, there are `Preferred exporter` and `Date` pages

**Comment by hungvu193** [comment]
We probably need to wait for this one right?
https://github.com/Expensify/App/issues/43532

**Comment by war-in** [comment]
Yes, I copied some stuff to make it work so we should wait 👍

**Comment by hungvu193** [comment]
Draft PR is up

**Comment by war-in** [comment]
I think [PR](https://github.com/Expensify/App/pull/43776) is ready but I'm waiting for backend to test it throughly. @hungvu193 are there any updates on the backend PR for the `export` page? 👀

**Comment by hungvu193** [comment]
I think @yuwenmemon or @NikkiWines can give you the BE update 👀

**Comment by hungvu193** [comment]
> I think [PR](https://github.com/Expensify/App/pull/43776) is ready but I'm waiting for backend to test it throughly. @hungvu193 are there any updates on the backend PR for the `export` page? 👀

Did you get the answer via slack?

**Comment by war-in** [comment]
@hungvu193 Yes! All questions were answered :)
tl;dr
Backend for `export` is almost ready (or ready). I'll check that soon

**Comment by hungvu193** [comment]
Awesome :)

**Comment by yuwenmemon** [comment]
Just merged the Export APIs they should be on staging on Monday

**Comment by hungvu193** [comment]
PR will be ready for review today.

**Comment by war-in** [comment]
@hungvu193 I think we should split my export PR because it grows quickly (it's over 1000 lines already).
I suggest adding offline and error states in a followup PR (I'll start working on it soon). 

Please share your thoughts here 🙏

**Comment by hungvu193** [comment]
> @hungvu193 I think we should split my export PR because it grows quickly (it's over 1000 lines already).
> 
> I suggest adding offline and error states in a followup PR (I'll start working on it soon). 
> 
> 
> 
> Please share your thoughts here 🙏 

Yes please go ahead 😁. Totally agree with you

**Comment by war-in** [comment]
Update: 
- https://github.com/Expensify/App/pull/43776 approved by @hungvu193 🔥 
- I'm waiting for the fixes in API

**Comment by hungvu193** [comment]
I'd suggest we should merge https://github.com/Expensify/App/pull/43776 first then test the API and other stuff in a follow PR (https://github.com/Expensify/App/pull/44739)

**Comment by hungvu193** [comment]
No payment needed for me here. Feel free to close this one.

**Comment by puneetlath** [comment]
Got it, thanks!
